src/ferny/ssh_askpass.py

```diff
@@ -0,0 +1,181 @@
+import logging
+import re
+from typing import ClassVar, Dict, Optional, Sequence
+
+from .interaction_agent import AskpassHandler
+
+logger = logging.getLogger(__name__)
+
+
+class AskpassPrompt:
+    stderr: str
+    messages: str
+    prompt: str
+
+    def reply(self, response: str) -> None:
+        pass
+
+    def close(self) -> None:
+        pass
+
+    async def handle_via(self, responder: 'SshAskpassResponder') -> None:
+        try:
+            response = await self.dispatch(responder)
+            if response is not None:
+                self.reply(response)
+        finally:
+            self.close()
+
+    async def dispatch(self, responder: 'SshAskpassResponder') -> Optional[str]:
+        return await responder.do_prompt(self)
+
+
+class SSHAskpassPrompt(AskpassPrompt):
+    ANSWERS: ClassVar[Optional[Sequence[str]]] = None
```

Comment:
Nitpicking: Could this drop the `Optional` and just default to `[]`?

Reply:
I'm allergic to adding mutable state at class initialization, so it would have to be `()`, but I think there's a distinction to be made here:

 - `None` means that there is no restriction on the list of valid answers.  This is the case for most prompts.
 - a sequence of items is a list of the valid responses; including the case of:
 - an empty sequence `()` is an indication that no answer is valid or expected.  That's used for FIDO "please touch your token" prompts that don't expect user input

Reply:
Probably the answer here is "this needs better documentation"

Reply:
Makes sense, thanks!